DNN Platform/Library/Entities/Modules/ModuleController.cs

```diff
@@ -250,6 +251,93 @@ private void CopyTabModuleSettingsInternal(ModuleInfo fromModule, ModuleInfo toM
             }
         }
 
+        /// <summary>
+        /// Checks whether module VIEW permission is inherited from its tab
+        /// </summary> 
+        /// <param name="module">The module</param>
+        /// <param name="permission">The module permission.</param>
+        private bool IsModuleViewPermissionInherited(ModuleInfo module, ModulePermissionInfo permission)
+        {
+            Requires.NotNull(module);
+
+            Requires.NotNull(permission);
+
+            var permissionViewKey = "VIEW";
+
+            if (!module.InheritViewPermissions || permission.PermissionKey != permissionViewKey)
+            {
+                return false;
+            }
+
+            var tabPermissions = TabPermissionController.GetTabPermissions(module.TabID, module.PortalID);
+
+            return tabPermissions
+                .ToList()
+                .Find(x => x.RoleID == permission.RoleID && x.PermissionKey == permissionViewKey) != null;

```

Comment:
Why does this need a `ToList`?  Couldn't we just use `Any()` to check the condition?

```
return tabPermissions.Any(p => p.RoleID == permission.RoleID && p.PermissionKey == permissionViewKey)
```

Reply:
`tabPermissions` is a `TabPermissionCollection` 

See: https://github.com/dnnsoftware/Dnn.Platform/blob/development/DNN%20Platform/Library/Security/Permissions/TabPermissionCollection.cs 

It does not implement `Any()` method. However, it can be easily added.

Reply:
Gotcha.  How about this:
```
return tabPermissions.Where(p => p.RoleID == permission.RoleID && p.PermissionKey == permissionViewKey).Any();
```